Add support for configuring OIDC client secrets (#28)
appuio/component-openshift4-authentication: docs/modules/ROOT/pages/how-tos/upgrade-2.x-to-3.x.adoc

= Upgrade from 2.x to 3.x

This guide describes the steps to perform an upgrade of the component from version 2.x to 3.x.

== Step-by step guide

> Version 3.0 adds support to configure the OIDC secret using vault instead of manually deploying the secret on the cluster.

Reviewer: ```suggestion
Version 3.0 adds support to configure the OIDC secret using Vault instead of having to manually deploy the secret on the cluster.
```

> To upgrade to the latest version follow these steps:

Reviewer: ```suggestion
To upgrade to version 3.x follow these steps:
```

. Extract the `clientSecret` from the `Secret` on the cluster.
The secret is referenced in the following parameter: `openshift4_authentication.identityProviders.<name_of_the_provider>.openID.clientSecret.name`

> . Put the secret in vault. For example at this location: `${cluster:tenant}/${cluster:name}/oidc/clientSecret`

Reviewer: ```suggestion
. Put the secret in Vault.
For example at this location: `${cluster:tenant}/${cluster:name}/oidc/clientSecret`
```

> . Change the `clientSecret` parameter as follows:
> [source]
> -          clientSecret:
> -            name: <name_of_the_secret>
> +          clientSecret: '?{vaultkv:${cluster:tenant}/${cluster:name}/oidc/clientSecret}'

Reviewer: ```suggestion
. Change the `clientSecret` parameter as follows:
+
[source,diff]
----
-clientSecret:
-  name: <name_of_the_secret>
+clientSecret: '?{vaultkv:${cluster:tenant}/${cluster:name}/oidc/clientSecret}' <1>
----
<1> Use the Vault location selected in the previous step here.
```
